zulipterminal/model.py

```diff
@@ -614,6 +617,57 @@ def get_other_subscribers_in_stream(self, stream_id: Optional[int]=None,
                     if stream['name'] == stream_name
                     if sub != self.user_id]
 
+    def get_user_info(self, user_id: int) -> Dict[str, Any]:
+        # TODO: add custom fields later as an enhancement
+        res_data = next((i for i in self.initial_data['realm_users']
+                         if i['user_id'] == user_id), None)
+        if not res_data:
+            # If the user is a cross realm bot (e.g., Welcome Bot)
+            res_data = next((i for i in self.initial_data['cross_realm_bots']
+                             if i['user_id'] == user_id), None)
+
+            # No such user with given user_id exists
+            if not res_data:
+                return {}
+
+        data = res_data
+        user_info = {}
+
+        user_info['Name'] = (data['full_name'] if data['full_name'] else
+                             '(No name)')
+        if data.get('email', None):
+            user_info['Email'] = data['email']
+        if data.get('date_joined', None):
+            user_info['Date joined'] = data['date_joined'][:10]
+        if data.get('timezone', None):
+            user_info['Timezone'] = data['timezone']
+            user_info['Local time'] = datetime.now(
+                pytz.timezone(str(data['timezone']))).strftime("%H:%M")
+
+        user_role = OrderedDict([
+            ('Owner', data.get('is_owner', False)),
+            ('Admin', data.get('is_admin', False)),
+            ('Guest', data.get('is_guest', False)),
+        ])  # type: OrderedDict[str, bool]
+
+        if data.get('is_bot', None):
+            user_info['Role'] = bot_types[data['bot_type']]
+            if data.get('bot_owner_id', None):
+                user_info['Owner'] = (self.user_name_from_id(
+                                      data['bot_owner_id']))
+
+        else:
+            user_info['Role'] = next((r for r, is_role in user_role.items()
+                                      if is_role), 'Member')
+
+            if self.initial_data['presences'].get(data['email'], None):
+                presence = (self.initial_data['presences'][data['email']]
+                            ['aggregated']['timestamp'])
+                user_info['Last active'] = (datetime.fromtimestamp(presence)
+                                            .strftime("%d-%b-%Y %H:%M:%S"))
```

Comment:
Certainly. I didn't rebase this branch for a while so was unaware of that method :sweat_smile: 

Are you suggesting shifting the `formatted_local_time` method in `helper`? If so then I might do this refactor in a prior commit? (Assuming the aspect of this commit unrelated)

>However, in general I'd suggest this method doesn't aim as much for priming the data just for "output", so much as doing error-checking and refining - so that time function could sit in the UI, potentially?

I'm unclear about which method you mean by `this method`. Could you clarify this a bit? :sweat_smile:

Reply:
Re the new/centralized method, the date/time output is centralized - and after #917 will be user-customized via the user account setting on the server - so the user likely should expect that format. So instead of using datetime directly, just use that new method? I'm not sure where helper comes into it?

Re this method, I meant the model one you're adding to get the user info - there are fine lines between the data from the server, the clarified/checked internal data, and that presented to the user. However, for now we can work on getting it working in a "reasonable" place and iterate from there - it's a bit like optimization: take care over a sensible starting point, but let's not debate for too long over how to improve it rather than getting it done :)

Reply:
Thanks for the clarification!